client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,169 @@
+     */
+    public TimeValue getWaitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * Sets the minimum time that the request should wait before returning a partial result (defaults to 1 second).
+     */
+    public void setWaitForCompletion(TimeValue waitForCompletion) {
+        this.waitForCompletion = waitForCompletion;
+    }
+
+    /**
+     * Returns whether the resource resource should be removed on completion or failure (defaults to true).
+     */
+    public Boolean isCleanOnCompletion() {
+        return cleanOnCompletion;
+    }
+
+    /**
+     * Determines if the resource should be removed on completion or failure (defaults to true).
+     */
+    public void setCleanOnCompletion(boolean cleanOnCompletion) {
+        this.cleanOnCompletion = cleanOnCompletion;
+    }
+
+    /**
+     * Get the amount of time after which the result will expire (defaults to 5 days).
+     */
+    public TimeValue getKeepAlive() {
+        return keepAlive;
+    }
+
+    /**
+     * Sets the amount of time after which the result will expire (defaults to 5 days).
+     */
+    public void setKeepAlive(TimeValue keepAlive) {
+        this.keepAlive = keepAlive;
+    }
+
+    // setters for request parameters of the wrapped SearchRequest
+    /**
+     * Set the routing values to control the shards that the search will be executed on.
+     * A comma separated list of routing values to control the shards the search will be executed on.
+     */
+    public void routing(String routing) {
+        this.searchRequest.routing(routing);
+    }
+
+    /**
+     * Set the routing values to control the shards that the search will be executed on.
+     */
+    public void routing(String... routings) {
+        this.searchRequest.routing(routings);
+    }
+
+    /**
+     * Sets the preference to execute the search. Defaults to randomize across shards. Can be set to
+     * {@code _local} to prefer local shards or a custom value, which guarantees that the same order
+     * will be used across different requests.
+     */
+    public void preference(String preference) {
+        this.searchRequest.preference(preference);
+    }
+
+    /**
+     * Specifies what type of requested indices to ignore and how to deal with indices wildcard expressions.
+     */
+    public void indicesOptions(IndicesOptions indicesOptions) {
+        this.searchRequest.indicesOptions(indicesOptions);
+    }
+
+    /**
+     * The search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public void searchType(SearchType searchType) {
+        this.searchRequest.searchType(searchType);
+    }
+
+    /**
+     * Sets if this request should allow partial results. (If method is not called,
+     * will default to the cluster level setting).
+     */
+    public void allowPartialSearchResults(boolean allowPartialSearchResults) {
+        this.searchRequest.allowPartialSearchResults(allowPartialSearchResults);
+    }
+
```

Comment:
We should also expose batchedReduceSize since that's the option to use to change the granularity of the progress ?

Reply:
Sure, I didn't add that because we implicitely set it to 5 in the ctor now, so thought its something we don't want the user to set.
What about PreFilterShardSize and requestCache then, those I also didn't add since we set them to fixed values internally?

Reply:
I would expose batchedReduceSize and requestCache only, preFilterShardSize is not necessary imo (we want to always run the can_match phase in async search).